partOfSpeech.ts

```diff
@@ -0,0 +1,217 @@
+import { hasProp } from './util';
+
+const clauses = {
+  'n..': 'noun',
+  'n.':  'noun',
+  n:     'noun',
+
+  'a. .':   'adjective',
+  'a.':     'adjective',
+  a:        'adjective',
+  'p. a.':  'past adjective', // ?? ?? ??
+  'p.a.':   'past adjective', //  ?? ?? ??
+  'pr. a.': 'proper adjective',
+  'adj.':   'adjective', // but why also just 'a.'?
+
+  'v.':    'verb',
+  'v. i.': 'intransitive verb',
+  'v. t.': 'transitive verb',
+
+  'imp.':       'imperfect (simple past tense)', // could also be imperative
+  'imp. sing.': 'singular imperfect (simple past tense)', // could also be imperative
+  'imp. pl.':   'plural imperfect (simple past tense)', // could also be imperative
+
+  'p. p.':              'past participle',
+  'past part.':         'past participle',
+  'p. p':               'past participle',
+  'imp. p. p.':         'imperfect past participle', // ?? ?? ??
+  'present participle': 'present participle',
+  'p. pr.':             'present participle',
+  'p. pr':              'present participle', // <3
+  'p, pr':              'present participle', // ?? ?? ??
+  'p pr.':              'present participle', // <3
+
+  'adv.': 'adverb',
+  adv:    'adverb',
+  'ads.': 'adverb', // ?
+
+  'superl.':   'superlative',
+  superl:      'superlative',
+  'comp.':     'comparative',
+  compar:      'comparative',
+  'compar.':   'comparative',
+  'prep.':     'preposition',
+  'interj.':   'interjection',
+  'pron.':     'pronoun',
+  'pron. pl.': 'plural pronoun',
+  'conj.':     'conjunction',
+
+  'third pers. sing. pres.': 'third-person singular present tense',
+  '3d. pers. sing. pres.':   'third-person singular present tense',
+  '3d pers. sing. pres.':    'third-person singular present tense',
+  '3d sing. pres.':          'third-person singular present tense',
+  '3d pers. sing. pr.':      'third-person singular present tense',
+  '3d sing. pr.':            'third-person singular present tense',
+  '3d pers. sing. present':  'third-person singular present tense',
+  '3d pers. pres.':          'third-person present tense',
+  '3d pers. sing.':          'third-person singular',
+  'imperative sing.':        'imperative singuar',
+  '2d pers. sing. imp.':     'second-person singular imperfect',
+  '2d pers. sing. pres.':    'second-person singular present tense',
```

Comment:
For my taste it goes a bit over on the balance between brevity <-> understandibility.
I think "tense" could be dropped in these lines